cmd/cli/main.go

```diff
@@ -14,6 +14,160 @@
 
 package main
 
-func main() {
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path"
+
+	"github.com/spf13/cobra"
+
+	"antrea.io/resource-auditing/pkg/types"
+)
+
+// tag flags
+var tagAuthor, tagEmail string
+
+// rollback flags
+var rollbackTag, rollbackSHA string
+
+// shared flags
+var serverAddr string
+
+var commandName = path.Base(os.Args[0])
+
+var rootCmd = &cobra.Command{
+	Use:  commandName,
+	Long: commandName + " is the command line tool for managing the auditing resource repository",
+}
+
+var tagCmd = &cobra.Command{
+	Use:   "tag create tag_name commit_sha [-a author] [-e email]\n   or: tag delete tag_name",
+	Short: "tags commits in the repository",
+	Args: func(cmd *cobra.Command, args []string) error {
+		if len(args) < 2 {
+			return fmt.Errorf("too few args")
+		}
```

Comment:
please look at the cobra helper functions to do this more elegantly. https://pkg.go.dev/github.com/spf13/cobra#MinimumNArgs in this case

Reply:
I'm a little unsure how to use these helper functions, since they return functions themselves and not an error on the condition not being satisfied

Reply:
hmm looks like it is indeed for very simple cases and there may not be a way to combine it with you own logic...